apps/server/src/modules/authentication/controllers/api-test/login.api.spec.ts

```diff
@@ -210,114 +210,104 @@ describe('Login Controller (api)', () => {
+
+				await em.persistAndFlush([system, school, studentRoles, user, account]);
+				em.clear();
+
+				const idToken: string = jwt.sign(
+					{
+						sub: 'testUser',
+						iss: system.oauthConfig?.issuer,
+						aud: system.oauthConfig?.clientId,
+						iat: Date.now(),
+						exp: Date.now() + 100000,
+						// For OIDC provisioning strategy
+						external_sub: userExternalId,
+					},
+					privateKey,
+					{
+						algorithm: 'RS256',
+					}
+				);
+
+				const axiosMock: MockAdapter = new MockAdapter(axios);
+
+				axiosMock.onPost(system.oauthConfig?.tokenEndpoint).reply<OauthTokenResponse>(200, {
+					id_token: idToken,
+					refresh_token: 'refreshToken',
+					access_token: 'accessToken',
+				});
+
+				return {
+					system,
+				};
+			};
 
 			it('should return jwt', async () => {
-				const params: OauthAuthorizationQueryParams = new OauthAuthorizationQueryParams();
-				params.code = 'someCode';
-
-				if (!system.oauthConfig) {
-					fail('oauth system not properly initialized');
-				}
+				const { system } = await setup();
 
-				const response = await request(app.getHttpServer()).get(`${basePath}/oauth/${system.id}`).query(params).send();
+				const response: Response = await request(app.getHttpServer())
+					.post(`${basePath}/oauth2`)
+					.send({
+						redirectUri: 'redirectUri',
+						code: 'code',
+						systemId: system.id,
+					})
+					.expect(HttpStatus.OK);
 
 				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-				expect(response.status).toEqual(302);
+				expect(response.body.accessToken).toBeDefined();
 			});
 		});
 
-		describe('when user login fails', () => {
-			it('should return error response', async () => {
-				const params: OauthAuthorizationQueryParams = new OauthAuthorizationQueryParams();
-				params.error = 'someCode';
+		describe('when an error is provided', () => {
+			const setup = async () => {
+				const schoolExternalId = 'schoolExternalId';
+				const userExternalId = 'userExternalId';
+
+				const system = systemFactory.withOauthConfig().buildWithId({});
+				const school = schoolFactory.buildWithId({ systems: [system], externalId: schoolExternalId });
+				const studentRoles = roleFactory.build({ name: RoleName.STUDENT, permissions: [] });
+				const user = userFactory.buildWithId({ school, roles: [studentRoles], externalId: userExternalId });
+				const account = accountFactory.buildWithId({
+					userId: user.id,
+					systemId: system.id,
+				});
 
-				if (!system.oauthConfig) {
-					fail('oauth system not properly initialized');
-				}
+				await em.persistAndFlush([system, school, studentRoles, user, account]);
+				em.clear();
 
-				const response = await request(app.getHttpServer()).get(`${basePath}/oauth/${system.id}`).query(params).send();
+				return {
+					system,
+				};
+			};
 
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-				expect(response.status).toEqual(302);
+			it('should return jwt', async () => {
```

Comment:
describe('when an error is provided', () => {
      it('should return internal server error.. ?